feat(Offer): add functionality to accept offer

nwjbrandon opened the pull request:
This patch includes the functionality for user to accept offer.
When the accept button is clicked the event is emitted to the
backend to indicate that the offer has been accepted. A socket
response is returned to indicate that the offer has successfully
been accepted. The buttons that show the offer in the conversation
will be hidden.

nwjbrandon committed:
feat(Offer): add functionality to accept offer

This patch includes the functionality for user to accept offer.
When the accept button is clicked the event is emitted to the
backend to indicate that the offer has been accepted. A socket
response is returned to indicate that the offer has successfully
been accepted. The buttons that show the offer in the conversation
will be hidden.

nwjbrandon committed:
style(Comments): Remove comments in ChatMessage

This patch remove the comments in ChatMessage. This comments are
just to help developer to know what the socket response is.

nwjbrandon committed:
fix(Offer): change relative imports to absolute imports

karrui commented on src/reducers/ChatDux.js:
```diff
@@ -30,8 +37,28 @@ const chat = createSlice({
+      );
+
+      // update chatConversation offer timestamp
+      // eslint-disable-next-line no-param-reassign
```
Disable the rule from the whole block maybe, now it's just cluttering the code.

```
/* eslint-disable no-param-reassign */
const chat = createSlice({
....
}
/* eslint-enable no-param-reassign */
```

Remove all `// eslint-disable-next-line no-param-reassign` comments.

karrui reviewed (approved):
LGTM

nwjbrandon committed:
fix(Offer): migrate redux store and remove disable param reassign

This patch migrate redux store to version 1 due to change in the
chat conversation model. The eslinting for param reassign is also
removed in ChatDux to prevent the code from being cluttered.
